### ISSUE TITLE
Rule proposal: `prefer-object-has-own`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1314]
Prefer Stage 3 proposal [Object.hasOwn()](https://github.com/tc39/proposal-accessible-object-hasownproperty) over `Object.prototype.hasOwnProperty.call()`

## Fail

```js
const hasProperty = Object.prototype.hasOwnProperty.call(object, property);
```

```js
const hasProperty = {}.hasOwnProperty.call(object, property);
```

## Pass

```js
const hasProperty = Object.hasOwn(object, property);
```

**Comment by yangmingshan** [comment]
Should we wait until this proposal reach to stage 4? That's [what ESLint does](https://github.com/eslint/eslint#what-about-experimental-features).

**Comment by sindresorhus** [comment]
@yangmingshan No, we implement things when we think they're likely to happened. That usually means stage 3 and up. ESLint is a bit too conservative. However, we disable these rules by default in the recommended preset.

**Comment by fregante** [comment]
FYI to anyone landing here: This has been added to the core:

- https://eslint.org/docs/latest/rules/prefer-object-has-own
